model/src/main/java/org/projectnessie/model/IcebergTableMetadata.java

```diff
@@ -0,0 +1,101 @@
+ */
+package org.projectnessie.model;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.immutables.value.Value;
+
+/**
+ * Represents the state of an Iceberg table in Nessie. An Iceberg table is globally identified via
+ * its {@link Content#getId() unique ID}.
+ *
+ * <p>The Iceberg-table-state consists of the location to the table-metadata and the state of
+ * relevant IDs using a serialized version of those.
+ *
+ * <p>When adding a new table (aka content-object identified by a content-id), use a {@link
+ * org.projectnessie.model.Operation.Put} without an expected-value. In all other cases (updating an
+ * existing table). always pass the last known version of {@link IcebergTable} as the expected-value
+ * within the put-operation.
+ */
+@Schema(
+    type = SchemaType.OBJECT,
+    title = "Iceberg table global state",
+    description =
+        "Represents the global state of an Iceberg table in Nessie. An Iceberg table is globally "
+            + "identified via its unique 'Content.id'.\n"
+            + "\n"
+            + "A Nessie commit-operation, performed via 'TreeApi.commitMultipleOperations', for Iceberg "
+            + "for Iceberg consists of a 'Operation.Put' with an 'IcebergTable' as in the 'content' "
+            + "field and the previous value of 'IcebergTable' in the 'expectedContent' field.\n"
+            + "\n"
+            + "During a commit-operation, Nessie checks whether the known global state of the "
+            + "Iceberg table is compatible (think: equal) to 'Operation.Put.expectedContent'.")
+@Value.Immutable
+@JsonSerialize(as = ImmutableIcebergTableMetadata.class)
+@JsonDeserialize(as = ImmutableIcebergTableMetadata.class)
+@JsonTypeName("ICEBERG_METADATA")
+public abstract class IcebergTableMetadata extends Content {
+
+  /**
+   * Location where Iceberg stored its {@code TableMetadata} file. The location depends on the
+   * (implementation of) Iceberg's {@code FileIO} configured for the particular Iceberg table.
+   */
+  @Nullable
+  public abstract String getMetadataLocation();
+
+  public abstract int getFormatVersion();
+
+  public abstract String getUuid();
+
+  public abstract String getLocation();
+
+  public abstract long getLastSequenceNumber();
+
+  public abstract long getLastUpdatedMillis();
+
+  public abstract int getLastColumnId();
+
+  public abstract String getSchema();
+
+  public abstract PartitionSpec getSpec();
+
+  public abstract SortOrder getSortOrder();
+
+  public abstract Snapshot getSnapshot();
+
+  public abstract Map<String, String> getProperties();
+
+  public abstract int getLastAssignedPartitionId();
```

Comment:
Shoud we make this ID (and last-seq-num, last-column-id) nullable instead of using Iceberg's "not present" sentinel? I'm fine with either though.